### DISCUSSION TITLE
i18n refactoring strategy

### DISCUSSION THREAD

**Discussion by agaqueiroz** [lbalazscs/Pixelitor#364] [General]
Hi @lbalazscs,

I saw that you are preparing changes for i18n in 1ad9490 commit. What's your strategy: are you changing to java.util.ResourceBundle?  are you mixing this with pixelitor.utils.Texts? or other strategy? 

Best regards,

angalq

**Comment by lbalazscs** [comment]
Hi André,

Thanks for the PR #363, I have merged it.

Regarding the tool settings i18n commit, you can check out the comments in the discussion #358. (If you're thinking about translating tool settings, you might want to sync up with @PatLabGit. Maybe adding the keys to the Java source code would be easier for you, since @PatLabGit doesn't have experience with Java)

As for the Texts vs ResourceBundle question: it doesn't really matter, because Texts is just a simple wrapper around the same ResourceBundle object, and I don't have any plans to change that. The advantage of Texts is that a translation key can be added with a single static call to Texts.i18n,  without having to pass around a ResourceBundle argument to different methods. The downside is that Texts isn't a standard class, and the Texts class must be initialized (including in all unit tests) before using it (in the past I had some bugs related to this initialization, but it should work now). So I used ResourceBundle whenever passing the object was easy, and Texts whenever passing a ResourceBundle was less convenient. If you add new keys using either of these methods, I'll happily merge them.

Regards,
László
